doc/content/getting-started/migrating-from-networks/device-json.md

```diff
@@ -6,93 +6,147 @@ weight: 1
-| `frequency_plan_id` | `defined_only` | e.g.  `EU_863_870`. [More info]({{< ref "reference/glossary#frequency-plan" >}}) |
-| `supports_join` | boolean | `true` for OTAA, `false` for ABP devices |
-| `root_keys.nwk_app.key` | uint128 | Application Key. [More info]({{< ref "reference/glossary#application-key" >}}) |
-| `root_keys.nwk_key.key` | uint128 | Network Key. Only for LoRaWAN version 1.1+ |
-| `mac_settings.rx1_delay.value` | `RxDelayValue` | Optional. Typical values are `RX_DELAY_1` (1 second) and `RX_DELAY_5` (5 seconds).  [More info]({{< ref "reference/api/end_device#message:MACSettings" >}})|
-| `mac_settings.supports_32_bit_f_cnt` | boolean | `true` for 32 bit frame counters, `false` for non-32 bit counters). [More info]({{< ref "reference/api/end_device#message:MACSettings" >}})  |
-| `session.dev_addr` | uint32 | Device Address. [More info]({{< ref "reference/glossary#devaddr" >}}) |
-| `session.keys.app_s_key.key` | uint128 | Application Session Key. [More info]({{< ref "reference/glossary#application-session-key" >}}) |
-| `session.keys.f_nwk_s_int_key.key` | uint128 | Forwarding Network Session Integrity Key, also referred to as **Network Session Key** in LoRaWAN v1.0.x compatibility mode. [More info]({{< ref "reference/api/end_device#message:SessionKeys" >}}) |
-| `session.last_f_cnt_up` | int | Optional, frame counter uplink. [More info]({{< ref "reference/api/end_device#message:MACSettings" >}}) |
-| `session.last_n_f_cnt_down` | int | Optional, frame counter downlinks. [More info]({{< ref "reference/api/end_device#message:MACSettings" >}}) |
-
-> Note: The dots in the **Field** column imply an embedded object. For example, `root_keys.nwk_app.key` must be set as: 
+Create a `devices.json` file containing a device object, like so:
+
+```json
+{
+  "ids": {
+    "device_id": "my-device",
+    "dev_eui": "0102030405060708",
+    "join_eui": "0102030405060708"
+  },
+  "name": "My Device",
+  "description": "Living room temperature sensor",
+  "lorawan_version": "MAC_V1_0_2",
+  "lorawan_phy_version": "PHY_V1_0_2_REV_B",
+  "frequency_plan_id": "EU_863_870_TTN",
+  "supports_join": true,
+  "root_keys": {
+    "app_key": {
+      "key": "01020304050607080102030405060708"
+    }
+  }
+}
+```
+
+You can define multiple end devices in a single file like so:
+
+```js
+[
+  {
+    /* device 1 */
+  },
+  {
+    /* device 2 */
+  }
+]
+```
+
+## JSON End Device Format
+
+The full specification of the JSON format is defined in the API protos, see the [EndDevice]({{< ref "/reference/api/end_device#message:EndDevice" >}}) message definition for details.
+
+The linked specification is quite extensive, and contains a lot of fields that are not required, or are only set and used internally by the Network Server. Below, the required and most commonly used fields are discussed.
+
+| Field | Required | Type | Example | Description |
```

Comment:
This table overflows the content column and isn't scrollable :( Plz create an issue to fix it and I'll look at it when I have time.

![localhost_1313_getting-started_migrating-from-networks_device-json_(Laptop with HiDPI screen)](https://user-images.githubusercontent.com/6963436/98126651-1e875980-1eb6-11eb-8322-6cab3b26b44d.png)